onnxruntime/core/providers/webgpu/webgpu_context.cc

```diff
@@ -144,6 +148,12 @@ void WebGpuContext::Initialize(const WebGpuBufferCacheConfig& buffer_cache_confi
       query_type_ = TimestampQueryType::None;
     }
   });
+
+#if defined(ENABLE_PIX_FOR_WEBGPU_EP)
+  // set pix frame generator
+  pix_frame_generator_ = std::make_unique<WebGpuPIXFrameGenerator>();
```

Comment:
it looks like the behavior is always creating the surface regardless of the config value of `WebGPU:enablePIXCapture`. Is this expected?

Reply:
use
```
pix_frame_generator_ = std::make_unique<WebGpuPIXFrameGenerator>(Instance(),
                                                                 Adapter(),
                                                                 Device());
```

so that:
1. no need to use friend class (which should be avoid)
2. no need to use Initialize